src/site/content/en/blog/optimize-long-tasks/index.md

```diff
@@ -0,0 +1,403 @@
+<figure>
+  {% Img src="image/jL3OLOhcWUQDnR4XjewLBx4e3PC3/8Bhl9Ilki4tM0aC1nfn8.png", alt="A single long task versus the same task broken up into shorter task. The long task is one large rectangle, whereas the chunked task is five smaller boxes which are collectively the same width as the long task.", width="800", height="242" %}
+  <figcaption>
+    A visualization of a single long task versus that same task broken up into five shorter tasks.
+  </figcaption>
+</figure>
+
+The reason this matters is because when tasks are broken up, the browser has more of an opportunity to respond to higher-priority work—including user interactions.
+
+<figure>
+  {% Img src="image/jL3OLOhcWUQDnR4XjewLBx4e3PC3/0yV0ynwW7FujIwvCbCxQ.png", alt="A depiction of how breaking up a task can facilitate a user interaction. At the top, a long task blocks an event handler from running until the task is finished. At the bottom, the chunked up task permits the event handler to run sooner than it otherwise would have.", width="800", height="448" %}
+  <figcaption>
+    A visualization of what happens to interactions when tasks are too long and the browser can't respond quickly enough to interactions, versus when longer tasks are broken up into smaller tasks.
+  </figcaption>
+</figure>
+
+In the preceding figure at the top, an event handler queued up by a user interaction had to wait for a single long task before it could run, delaying the interaction. At the bottom, the event handler has a chance to run sooner. Because that event handler had an opportunity to run in between smaller tasks, it runs sooner than if it had to wait for one big task to complete. In the top example, the user might have noticed lag; in the bottom, the interaction might have felt _instant_.
+
+The problem, however, is that the typical advice of "break up your long tasks" and "don't block the main thread" isn't specific enough, unless you already know _how_ to do those things. That's exactly what this guide will explain.
+
+## Task management strategies
+
+One of the best pieces of advice you can follow in software architecture is to break up your work into little functions that do discrete things. This gives you the benefits of better code readability, project maintainability, and makes tests easier to write.
+
+```js
+function saveSettings () {
+  validateForm();
+  showSpinner();
+  saveToDatabase();
+  updateUI();
+  sendAnalytics();
+}
+```
+
+In this example, there's a function named `saveSettings` that calls five functions within it to do the work, such as validating a form, showing a spinner, sending data, and so on. Conceptually, this is well architected. If you need to debug one of these functions, you can traverse a project tree to figure out what each function does.
+
+The problem, however, is that JavaScript doesn't run each of these functions as separate tasks because they are being executed within the `saveSettings` function. **This means that all five functions run as a single task.**
```

Comment:
```suggestion
The problem, however, is that JavaScript doesn't run each of these functions as separate tasks because they are being executed within the `saveSettings()` function. **This means that all five functions run as a single task.**
```